src/test-reader/test-parser.ts

```diff
@@ -153,15 +153,15 @@ export class TestParser extends EventEmitter {
             treeBuilder.addTestFilter((test: Test) => {
                 let current: Test | Suite | null = test;
 
+                const allTags = new Set<string>([]);
+
                 while (current) {
-                    if (tag(current.tags)) {
-                        return true;
-                    } else {
-                        current = current.parent;
-                    }
+                    [...current.tags.keys()].forEach(item => allTags.add(item));
```

Comment:
why keys()? I know it's an alias in set for values, or we actually use Map for `tags`? It's becoming hard to follow, i think we should use either Map or Set everywhere for tags.

Reply:
answered above